src/types/utils.ts

```diff
@@ -1,2 +1,26 @@
-// Helper to extract type T from an array of type T
+/** Helper to extract the type of a single element in an array of elements */
 export type Unpacked<T> = T extends (infer U)[] ? U : T;
+
+/** Helper that takes in an object of key/value pairs and returns an object of key/value pairs where only one value is required */
+export type RequireOnlyOne<T, Keys extends keyof T = keyof T> = Pick<
+  T,
+  Exclude<keyof T, Keys>
+> &
+  {
+    [K in Keys]-?: Required<Pick<T, K>> &
+      Partial<Record<Exclude<Keys, K>, undefined>>;
+  }[Keys];
+
+/**  Helper that takes in an object of key/value pairs and returns an object of key/value pairs where at least one value is required */
+export type RequireAtLeastOne<T, Keys extends keyof T = keyof T> = Pick<
```

Comment:
I don't see this type imported anywhere but we could probably use this in a future test

Reply:
yup i added these types since they seemed to be something we could reuse in the future.